pkg/client/custom_api/client.go

```diff
@@ -142,13 +143,13 @@ func (mc *metricsClient) GetMetricByName(name types.NamespacedName, info provide
 }
 
 func (mc *metricsClient) GetMetricBySelector(namespace string, selector labels.Selector, info provider.CustomMetricInfo, metricSelector labels.Selector) (*custom_metrics.MetricValueList, error) {
-	var objects *v1beta2.MetricValueList
+	var objects *customMetricsAPI.MetricValueList
 	var err error
 	kind, err := mc.mapper.ResourceSingularizer(info.GroupResource.Resource)
 	if err != nil {
 		return nil, fmt.Errorf("failed to singularize %s: %v", info.GroupResource.Resource, err)
 	}
-	klog.Infof("custom metric info: %#v", info)
+	logger.V(1).Info("GetMetricBySelector", "metric_info", info.String())
```

Comment:
I can control the verbosity of the k8s API runtime using `--v` but it does not seem to affect this logger. I had a look at `klog` and I realised that it supports structured logging, should we not just reuse it?

```suggestion
	klog.V(1).InfoS("metricsFor", "selector", selector, "metric", info.String())
```

Reply:
I see the problem. The logger is used globally at the package level (`var logger = log.ForPackage("custom_api")`) before being configured. I fixed this in 54c3cd5.

Now that it works I reverted most of log to the debug level (`V(1)`), otherwise it seems too noisy.